nominal/cli/dataset.py

```diff
@@ -64,3 +67,45 @@ def get(rid: str, base_url: str, token: str | None) -> None:
     client = get_client(base_url, token)
     dataset = client.get_dataset(rid)
     print(dataset)
+
+
+@dataset_cmd.command("summarize")
+@click.option("-r", "--rid", required=True, multiple=True, help="RID(s) of the dataset(s) to summarize")
+@click.option(
+    "-c",
+    "--csv",
+    type=click.Path(exists=False, dir_okay=False, resolve_path=True, path_type=pathlib.Path),
+    help="If provided, a path to write the description to as a CSV",
+)
+@click.option("--show-rids", is_flag=True, help="If provided, show channel / dataset RIDs as part of tabulated output")
+@BASE_URL_OPTION
+@TOKEN_OPTION
+def summarize(rid: Sequence[str], csv: pathlib.Path | None, show_rids: bool, base_url: str, token: str | None) -> None:
+    """Summarize the dataset(s) by their schema (column names, types, and RIDs)"""
+
+    client = get_client(base_url, token)
+    datasets = {dataset.name: dataset for dataset in client.get_datasets(rid)}
```

Comment:
nit: i guess this dict isn't super useful since you just iterate over both and access the dataset object already?